### PR TITLE
Skip to content

### PR DESCRIPTION
**Pull request by discombobulateme**
# Description

We added a Skip to Main Content component.

## More Details

This is to ensure readability for screen readers and keyboard users.

## Corresponding Issue

https://github.com/ifmeorg/ifme/issues/1703

# Screenshots

Screen Shot 2020-04-17 at 20 00 02
Screen Shot 2020-04-17 at 20 00 13

---

Reviewing this pull request? Check out our [Code Review Practices](https://github.com/ifmeorg/ifme/wiki/Code-Review-Practices) guide if you haven't already!

### CODE REVIEW

File: client/app/components/SkipToContent/SkipToContent.scss [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  width: 1px;
+  height: 1px;
+  overflow: hidden;
+}
+ 
+.skipToContent:focus {
+  position: static;
+  width: auto;
+  height: auto;
+  color: #990019;
```
Comment (julianguyen) [inline client/app/components/SkipToContent/SkipToContent.scss:16]: Let's use the global variable for this colour i.e. `$carmine` instead.

File: client/app/components/SkipToContent/SkipToContent.scss [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  background: #91D7E8;
```
Comment (julianguyen) [inline client/app/components/SkipToContent/SkipToContent.scss:17]: Let's use the global variable for this colour i.e. `$cornflower` instead.

File: client/app/components/SkipToContent/index.jsx [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import React from 'react';
+import css from './SkipToContent.scss';
+
+export type Props = {
+  id?: string,
+};
+
+function SkipToContent(props: Props) {
+
+  return <a className={css.skipToContent} href={`#${props.id}`}>Skip to main content</a>;
```
Comment (julianguyen) [inline client/app/components/SkipToContent/index.jsx:11]: Since our app supports internationalization (i18n) we'll want to add "Skip to main content" as a key in `config/locales/en.yml` and translate it in other languages too. You can use Google translate here for now.

Then you can use 

```
import { I18n } from '../../libs/i18n';

// etc.

I18n.t('navigation.skip_to_main_content'); 

```

File: client/app/components/SkipToContent/index.jsx [hunk 4]
```diff
@@ -0,0 +1,15 @@
+// @flow
+import React from 'react';
+import css from './SkipToContent.scss';
+
+export type Props = {
+  id?: string,
```
Comment (julianguyen) [inline client/app/components/SkipToContent/index.jsx:6]: I would make this prop required.

Comment (julianguyen) [reply client/app/components/SkipToContent/index.jsx:7]: So this issue is responsible for the following Flow error that's showing up in the CircleCI build:

```
Cannot coerce `props.id` to string because undefined [1] should not be coerced.
 

 
   app/components/SkipToContent/index.jsx:13:53
 
   13|   return <a className={css.skipToContent} href={`#${props.id}`}>Skip to main content</a>;
 
                                                           ^^^^^^^^
 

 
References:
 
   app/components/SkipToContent/index.jsx:7:8
 
    7|   id?: string,
 
              ^^^^^^ [1]
 
```

**Comment by discombobulateme** [comment]
Dear @julianguyen and @nshki, I think @Victoria-Hodder and I finally got this!!!! 👊🏼 how do we PR this PR? 🧐 to code to code review again?

File: client/package.json [hunk 5]
```diff
@@ -49,7 +49,8 @@
     "react-html-parser": "^2.0.2",
     "react-lazyload": "^2.6.2",
     "react-on-rails": "11.3.0",
-    "react-render-html": "^0.6.0"
```
Comment (julianguyen) [inline client/package.json:52]: We can undo these changes since we're not using these dependencies. Same goes for the `yarn.lock` changes.

File: yarn.lock [hunk 6]
```diff
@@ -2,3 +2,179 @@
 # yarn lockfile v1
 
 
+"@babel/runtime@^7.1.2", "@babel/runtime@^7.4.0":
```
Comment (julianguyen) [inline yarn.lock:5]: Let's undo these changes!

File: client/app/components/SkipToContent/index.jsx [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+export type Props = {
+  id: string,
+};
+
+function SkipToContent(props: Props) {
+  const { id } = props;
+
+  window.addEventListener('Props', () => {
+    window.scrollTo(0, 0)});
```
Comment (julianguyen) [inline client/app/components/SkipToContent/index.jsx:15]: I think we're missing a `}` here.

I'm not sure what listening to "Props" actually does. I would assume it's listening to all prop changes? I don't think we want to do that.

Instead, I think we want to put an `onClick` listener on the link element below and put the scroll logic there.

File: client/package.json (hunk 5 above)
Comment (discombobulateme) [reply client/package.json:52]: Dear Julia, I will be very glad to undo it but I guess they came automatically from when I pulled updates from Master 😅. I wouldn't know how to und them 🤨. Any help?

Comment (julianguyen) [reply client/package.json:52]: You'll want to do a soft reset on the commit that has those changes and remove them from it. If that commit isn't the latest one. You can just cherry-pick the commits that are on top of that one.

Alternatively, you could just squash everything in one commit and do the soft reset on that.

File: client/app/components/SkipToContent/index.jsx (hunk 7 above)
Comment (Victoria-Hodder) [reply client/app/components/SkipToContent/index.jsx]: Hey @julianguyen ! Thanks for the feedback. We started working on that this week.

Would you like us to refactor our current SkipToContent function into a class component (in order to add the scroll logic)? Or could we also use react hooks? 

I'm asking because we're not sure if react hooks can be used in a react on rails context. 

Thank you!

Comment (julianguyen) [reply client/app/components/SkipToContent/index.jsx]: You can use React Hooks here! But if you feel more comfortable using React classes, that's fine too.

File: client/app/startup/registration.js [hunk 8]
```diff
@@ -18,7 +18,12 @@ import { Logo } from '../components/Logo';
 import { Modal } from '../components/Modal';
 import { Notifications } from '../widgets/Notifications';
 import { Resource } from '../components/Resource';
+<<<<<<< HEAD
```
Comment (julianguyen) [inline client/app/startup/registration.js:21]: Looks like this merge conflict was not resolved.

File: client/package.json (hunk 5 above)
Comment (Victoria-Hodder) [reply client/package.json:52]: Could we also use `git revert`?
From what I understand `reset` would lose the later commits and could lead to some conflicts with the memory/potential loss of code.
